pennylane/templates/subroutines/qsvt.py

```diff
@@ -150,6 +163,238 @@ def qsvt(A, angles, wires, convention=None):
+
+            # P(x) = -1 + 0.2 x^2 + 0.5 x^4
+            poly = np.array([-1, 0, 0.2, 0, 0.5])
+
+            hamiltonian = qml.dot([0.3, 0.4, 0.3], [qml.Z(2), qml.X(2) @ qml.Z(3), qml.X(2)])
+
+            dev = qml.device("default.qubit")
+
+            @qml.qnode(dev)
+            def circuit():
+                qml.qsvt(hamiltonian, poly, encoding_wires=[0, 1], block_encoding="prepselprep")
+                return qml.state()
+
+
+            matrix = qml.matrix(circuit, wire_order=[0, 1, 2, 3])()
+
+        .. code-block:: pycon
+
+            >>> print(np.round(matrix[:4, :4], 4).real)
+            [[-0.7158  0.      0.      0.    ]
+             [ 0.     -0.975   0.      0.    ]
+             [ 0.      0.     -0.7158  0.    ]
+             [ 0.     -0.      0.     -0.975 ]]
+
+
+        Alternatively, if the input ``A`` is a matrix, the valid values for ``block_encoding`` are
+        ``"embedding"`` and ``"fable"``. In this case, the ``encoding_wires`` parameter corresponds to
+        the ``wires`` attribute in the templates :class:`~pennylane.BlockEncode` and :class:`~pennylane.FABLE`,
+        respectively. Note that for QSVT to work, the input matrix must be Hermitian.
+
+        .. code-block:: python
+
+            # P(x) = -1 + 0.2 x^2 + 0.5 x^4
+            poly = np.array([-0.1, 0, 0.2, 0, 0.5])
+
+            A = np.array([[-0.1, 0, 0, 0.1], [0, 0.2, 0, 0], [0, 0, -0.2, -0.2], [0.1, 0, -0.2, -0.1]])
+
+            dev = qml.device("default.qubit")
+
+            @qml.qnode(dev)
+            def circuit():
+                qml.qsvt(A, poly, encoding_wires=[0, 1, 2, 3, 4], block_encoding="fable")
+                return qml.state()
+
+            matrix = qml.matrix(circuit, wire_order=[0, 1, 2, 3, 4])()
+
+        .. code-block:: pycon
+
+            >>> print(np.round(matrix[:4, :4], 4).real)
+            [[-0.0954  0.     -0.0056 -0.0054]
+             [ 0.     -0.0912 -0.     -0.    ]
+             [-0.0056  0.     -0.0788  0.0164]
+             [-0.0054 -0.      0.0164 -0.0842]]
+
+        Note that for the FABLE block encoding to function correctly, it must comply with the following:
+
+        .. math::
+
+                d \|A\|^2 \leq 1,
+
+        where :math:`d` is the maximum dimension of :math:`A` and :math:`\|A\|` is the 2-norm of :math:`A`.
+        In the previous example this is satisfied since :math:`d = 4` and :math:`\|A\|^2 = 0.2`:
+
+        .. code-block:: pycon
+
+            >>> print(4* np.linalg.norm(A, ord='fro')**2)
+            0.8000000000000004
+
+
+    """
+
+    if encoding_wires is None or block_encoding is None or "wires" in kwargs.keys():
+        warnings.warn(
+            "You could be using the old `qsvt` functionality (now `qml.qsvt_legacy`). Make sure you pass a "
+            "polynomial instead of angles. Set a value for `block_encoding` if you want to silent this warning.",
+            qml.PennyLaneDeprecationWarning,
+        )
+
+    angles = qml.poly_to_angles(poly, "QSVT")
```

Comment:
I think we need to perform the input validation for `A` before computing the angles.
While the `"root-finding"` angle solver is quick, additional angle solver methods we may add in the future will likely have longer runtimes, and the user shouldn't have to wait for the angles to be computed just to get an error regarding `A`.

Reply:
Good point 🙌 Updated